pkg/cmd/sourcemapdiff/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/fatih/color"
+	"github.com/grafana/plugin-validator/pkg/cmd/sourcemapdiff/difftool"
+	"github.com/grafana/plugin-validator/pkg/sourcemap"
+)
+
+var options = map[string]*string{
+	"archiveUri":     flag.String("archiveUri", "", "The URI of the plugin archive. This can be a local file path (zip or folder) or a URL. If it's a URL, it must be a zip file."),
+	"sourceCodeUri":  flag.String("sourceCodeUri", "", "The URI of the source code. This can be a local file path (zip or folder) or a URL. If it's a URL, it must be a git repository or a zip file."),
+	"nonInteractive": flag.String("nonInteractive", "n", "If set to 'y', the command will not ask for user input and will use the default values for all the options."),
+}
+
+func main() {
+	flag.Parse()
+
+	// resolve the pluginPath from options
+	pluginPath, archiveCleanup, err := getLocalPathFromArchiveOption(*options["archiveUri"])
+	defer archiveCleanup()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// resolve the sourceCodePath from options
+	sourceCodePath, sourceCodeCleanup, err := getLocalPathFromSourceCodeOption(*options["sourceCodeUri"])
+	defer sourceCodeCleanup()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// extract the source map into a temporal folder
+	sourceCodeMapPath := filepath.Join(pluginPath, "module.js.map")
+
+	// compare the source map with the source code
+	report, err := difftool.CompareSourceMapToSourceCode(sourceCodeMapPath, filepath.Join(sourceCodePath, "src"))
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Println(report.GeneratePrintableReport())
+	if report.TotalDifferences > 0 && *options["nonInteractive"] == "n" {
+		promptToSeeDiff(sourceCodeMapPath, sourceCodePath)
+	}
+
+}
+
+func promptToSeeDiff(sourceCodeMapPath string, sourceCodePath string) {
+
+	var answer string
+	fmt.Println("Do you want to see the differences in your diff tool?")
+	fmt.Println("Be aware the original source code will contain more files than the source map such as images, readme files and typescript typefiles.")
+	fmt.Println("\n\nIt is recommended to install meld (https://meldmerge.org/) to see the differences.")
+	fmt.Print("Open diff tool? (y/n): ")
+	fmt.Scanln(&answer)
+
+	if (answer == "y") || (answer == "Y") {
+
+		systemDiffTool, err := getSystemDiffTool()
+		if err != nil {
+			color.Red(err.Error())
+			suggestDiffTool()
+			return
+		}
+
+		sourceCodeMapPath, err := sourcemap.ExtractSourceMapToPath(sourceCodeMapPath)
+		if err != nil {
+			fmt.Println("Error extracting source map to file system")
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		command := fmt.Sprintf(systemDiffTool, sourceCodeMapPath, filepath.Join(sourceCodePath, "src"))
+
+		// run the command
+		fmt.Println("Running command: ", command)
+		cmd := exec.Command("sh", "-c", command)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
```

Comment:
I don't really care about this possible error since some diff tools will return code 1 for no reason (e.g. the user didn't save the session).

I can't even be sure the command will return after the process dies because some diff tools will spawn new processes for the UI and return the cli